internal/pkg/runtime/launcher/oci/spec_linux.go

```diff
@@ -115,3 +118,25 @@ func addNamespaces(spec specs.Spec, ns launcher.Namespaces) specs.Spec {
 
 	return spec
 }
+
+// addCDIDevice adds the requested device to an existing spec.
+func addCDIDevice(spec specs.Spec, cdiDevice string) (specs.Spec, error) {
```

Comment:
Please could you move this into a new `cdi_linux.go` file here. It's just a preference for separating stuff a bit more as this launcher code is going to see a lot of work, and expansion for 4.0.